src/lib/datepicker/datepicker-toggle.ts

```diff
@@ -36,7 +36,7 @@ export class MdDatepickerToggle<D> {
   constructor(public _intl: MdDatepickerIntl) {}
 
   _open(event: Event): void {
-    if (this.datepicker) {
+    if (this.datepicker && !this.datepicker.disabled) {
```

Comment:
add your own disabled input:

```ts
@Input()
get disabled() {
  return this._disabled === undefined ? this.datepicker.disabled : this._disabled;
}
set disabled(value) {
  this._disabled = coerceBooleanValue(value);
}
```

because buttons also have a native disabled attribute you'll have to add the host binding too: `'[disabled]': 'disabled'`